File: frontend/src/api/styleReferences.ts
import { client } from "./client";
import type { GenerateJobAck, StyleReferenceKind } from "@/types/api";

const GENERATE_TIMEOUT_MS = 60_000;

export const styleReferencesApi = {
  generateStyleReference(projectId: string, kind: StyleReferenceKind): Promise<GenerateJobAck> {
    const path =
      kind === "character"
        ? `/projects/${projectId}/character-style-reference/generate`
        : `/projects/${projectId}/scene-style-reference/generate`;
    return client
      .post(path, undefined, { timeout: GENERATE_TIMEOUT_MS })
      .then((r) => r.data as GenerateJobAck);
  }
};

File: backend/app/api/style_references.py
from fastapi import APIRouter, Depends, Path
from sqlalchemy.ext.asyncio import AsyncSession

from app.api.envelope import Envelope
from app.deps import get_db
from app.domain.schemas.style_reference import StyleReferenceJobAck
from app.domain.services.style_reference_service import StyleReferenceService


router = APIRouter(prefix="/projects/{project_id}", tags=["style-references"])


@router.post("/character-style-reference/generate", response_model=Envelope[StyleReferenceJobAck])
async def generate_character_style_reference(
    project_id: str = Path(..., description="项目 ID"),
    db: AsyncSession = Depends(get_db),
):
    job = await StyleReferenceService(db).generate(project_id, "character")
    return Envelope.success(StyleReferenceJobAck(job_id=job.id))


@router.post("/scene-style-reference/generate", response_model=Envelope[StyleReferenceJobAck])
async def generate_scene_style_reference(
    project_id: str = Path(..., description="项目 ID"),
    db: AsyncSession = Depends(get_db),
):
    job = await StyleReferenceService(db).generate(project_id, "scene")
    return Envelope.success(StyleReferenceJobAck(job_id=job.id))
